sdks/python/apache_beam/io/gcp/gcsfilesystem.py

```diff
@@ -377,3 +377,17 @@ def report_lineage(self, path, lineage, level=None):
       # bucket only
       components = components[:-1]
     lineage.add('gcs', *components)
+
+  def check_splittability(self, path):
+    try:
+      file_metadata = self._gcsIO()._status(path)
+      if file_metadata.get('content_encoding', None) == 'gzip':
```

Comment:
Doesn't the content-type also have to be a particular thing in addition to the content-encoding being set to gzip?